### PR TITLE
Goa 2331 configurable cors

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    @Test
+    public void checkAllowedOriginsWereOverriden() throws IOException, ServletException {
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_ORIGIN), is(DEFAULT_ACCESS_CONTROL_ALLOW_ORIGIN));
+    }
+
+    @Test
+    public void checkAllowedCredentialsWereOverriden() throws IOException, ServletException {
+        MockHttpServletResponse response = new MockHttpServletResponse();
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_CREDENTIALS), is(DEFAULT_ACCESS_CONTROL_ALLOW_CREDENTIALS));
+    }
+
+    @Test
+    public void checkAllowedHeadersWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_HEADERS), is(DEFAULT_ACCESS_CONTROL_ALLOW_HEADERS));
+    }
+
+    @Test
+    public void checkAllowedMethodsWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_METHODS), is(DEFAULT_ACCESS_CONTROL_ALLOW_METHODS));
+    }
+
+    @Test
+    public void checkExposedHeadersWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_EXPOSE_HEADERS), is(nullValue()));
+    }
+
+    @Test
+    public void checkMaxAgeWasOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java (hunk 4 above)
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: Should this method be called ....NotOverridden?

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+@ActiveProfiles("cors-test")
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = CORSFilterReadingPropertiesTest.FakeApplication.class)
+public class CORSFilterReadingPropertiesTest {
+    @Autowired
+    private CORSFilter filter;
+
+    @Test
+    public void checkAllowedOriginsWereOverriden() throws IOException, ServletException {
+        MockHttpServletResponse response = new MockHttpServletResponse();
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_ORIGIN), is("my-origin"));
+    }
+
+    @Test
+    public void checkAllowedCredentialsWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_HEADERS), is("my-headers"));
+    }
+
+    @Test
+    public void checkAllowedHeadersWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_CREDENTIALS), is("my-credentials"));
+    }
+
+    @Test
+    public void checkAllowedMethodsWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_METHODS), is("my-methods"));
+    }
+
+    @Test
+    public void checkExposedHeadersWereOverriden() throws IOException, ServletException {
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        filter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        assertThat(response.getHeader(ACCESS_CONTROL_EXPOSE_HEADERS), is("my-exposed-headers"));
+    }
+
+    @Test
```
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Overriden ->Overridden

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java (hunk 8 above)
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Incorrect method name: Headers -> Credentials

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java (hunk 7 above)
Comment (TonyWardell) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: Incorrect file name Credentials -> Headers

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java (hunk 5 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: done

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java (hunk 4 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: yes ta

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java (hunk 1 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: n/a (but applicable now in the `CORSFilterReadPropertiesTest`)

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java (hunk 3 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterDefaultPropertiesTest.java]: n/a (but applicable now in the `CORSFilterReadPropertiesTest`)

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java (hunk 8 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: done

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java (hunk 9 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: done

File: rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java (hunk 11 above)
Comment (eddturner) [inline rest-common/src/test/java/uk/ac/ebi/quickgo/rest/controller/CORSFilterReadingPropertiesTest.java]: done

**Pull request merged**